pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test: TestLogic_mixed_version_upgrade_preserve_ttl failed

pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test.TestLogic_mixed_version_upgrade_preserve_ttl [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12879435?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12879435?buildTab=artifacts#/) on master @ [04a04ca61e6d3368689d685a86dae58b4ce464bf](https://github.com/cockroachdb/cockroach/commits/04a04ca61e6d3368689d685a86dae58b4ce464bf):


```
=== RUN   TestLogic_mixed_version_upgrade_preserve_ttl
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/5cf022a3d4669c7cd3127fd7792ab475/logTestLogic_mixed_version_upgrade_preserve_ttl1263059356
    test_log_scope.go:81: use -show-logs to present logs inline
    logic.go:2915: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8511/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test_/cockroach-go-testserver-upgrade-to-master_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl:20: SELECT version LIKE '%23.2-%' FROM [SHOW CLUSTER SETTING version]
        expected:
            true
            
        but found (query options: "retry") :
            false
            
[18:56:27] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8511/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test_/cockroach-go-testserver-upgrade-to-master_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl: 3 statements
[18:56:27] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8511/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test_/cockroach-go-testserver-upgrade-to-master_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl with config cockroach-go-testserver-upgrade-to-master: 3 tests, 1 failures
    logic.go:4005: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8511/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test_/cockroach-go-testserver-upgrade-to-master_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl:25: error while processing
    logic.go:4005: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8511/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test_/cockroach-go-testserver-upgrade-to-master_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl:25: too many errors encountered, skipping the rest of the input
cockroach logs captured in: /artifacts/tmp/_tmp/5cf022a3d4669c7cd3127fd7792ab475/cockroach-logs918162506
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5cf022a3d4669c7cd3127fd7792ab475/logTestLogic_mixed_version_upgrade_preserve_ttl1263059356
--- FAIL: TestLogic_mixed_version_upgrade_preserve_ttl (99.75s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_mixed_version_upgrade_preserve_ttl.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33928

pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test.TestLogic_mixed_version_upgrade_preserve_ttl [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12907668?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12907668?buildTab=artifacts#/) on master @ [ccc3d549aa7eacb6eff1ef8137a82a69acab291c](https://github.com/cockroachdb/cockroach/commits/ccc3d549aa7eacb6eff1ef8137a82a69acab291c):


```
=== RUN   TestLogic_mixed_version_upgrade_preserve_ttl
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/5cf022a3d4669c7cd3127fd7792ab475/logTestLogic_mixed_version_upgrade_preserve_ttl1657679936
    test_log_scope.go:81: use -show-logs to present logs inline
    logic.go:2915: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10494/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test_/cockroach-go-testserver-upgrade-to-master_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl:20: SELECT version LIKE '%23.2-%' FROM [SHOW CLUSTER SETTING version]
        expected:
            true
            
        but found (query options: "retry") :
            false
            
[02:20:30] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10494/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test_/cockroach-go-testserver-upgrade-to-master_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl: 3 statements
[02:20:30] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10494/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test_/cockroach-go-testserver-upgrade-to-master_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl with config cockroach-go-testserver-upgrade-to-master: 3 tests, 1 failures
    logic.go:4005: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10494/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test_/cockroach-go-testserver-upgrade-to-master_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl:25: error while processing
    logic.go:4005: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/10494/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/cockroach-go-testserver-upgrade-to-master/cockroach-go-testserver-upgrade-to-master_test_/cockroach-go-testserver-upgrade-to-master_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl:25: too many errors encountered, skipping the rest of the input
cockroach logs captured in: /artifacts/tmp/_tmp/5cf022a3d4669c7cd3127fd7792ab475/cockroach-logs2121529660
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5cf022a3d4669c7cd3127fd7792ab475/logTestLogic_mixed_version_upgrade_preserve_ttl1657679936
--- FAIL: TestLogic_mixed_version_upgrade_preserve_ttl (124.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_mixed_version_upgrade_preserve_ttl.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl
+++ b/pkg/sql/logictest/testdata/logic_test/mixed_version_upgrade_preserve_ttl
@@ -15,10 +15,11 @@ CREATE TABLE tbl (
 
 upgrade all
 
-# Verify that the cluster version upgrades have begun. Note that the first
-# cluster upgrade is the one that repairs all descriptors.
+# Verify that the cluster version upgrades have begun by asserting we're no
+# longer on the previous version. Note that the first cluster upgrade is the
+# one that repairs all descriptors.
 query B retry
-SELECT version LIKE '%23.2-%' FROM [SHOW CLUSTER SETTING version]
+SELECT version != '23.1' FROM [SHOW CLUSTER SETTING version]
 ----
 true
 

